### ISSUE TITLE
Linters to make applications more accessible

### ISSUE THREAD

**Issue by csantacruza** [dart-lang/sdk#58251]
** Describe the rule you would like to see implemented**

The purpose of this rule is to give some introductions to the world of accessibility to developers and the option of creating applications and tools accessible to all people.
The idea is to create a set of rules to make applications more accessible by following the Web Content Accessibility Guidelines (WCAG) 2.1

Set of rules to help people with visual and learning disabilities:

1. Text font size> = 12 (this value depends on the size of the device)
2. Semantic text label! = Null
3. Where possible use fonts from the San Serif family such as Arial, Roboto, Comic Sans, Helvetica, or Verdana. Or use alternative fonts like Tahoma, Century Gothic, and Trebuchet MS.
4. Semantic label for images, icons, and buttons.
5. Text and background contrast (more details in the guide)

Set of rules to help people with involuntary movements:

1. Padding on buttons for people with big fingers and involuntary movements

*Very important*
Show in the lint message that it is possible to use the showSemanticsDebugger property in the MaterialApp widget to view the labels in the application. And use Accessibility> TalkBack on Android or Accessibility> VoiceOver on iOS to test the tags.

MaterialApp (
            showSemanticsDebugger: true,
);

And prove that the semantics widget set for accessibility exists.

** Examples**

============ Text Font Size ================
**Bad**
Text(
   '...',
   style: TextStyle (fontSize: 8),
);

**Prefer**
Text('...'); // By default, font size = 14
or
Text(
   '...',
   style: TextStyle (fontSize: 12),
);

============ TextSemantic Tag ===============

By default Text ('...'); SemanticsLabel is equal to the value of the text.

**Prefer**
Text(
   '...',
  semanticsLabel: 'Description of this text',
);

============ Images ===============

**Optional**

By default, image and image widget. (active, network, file) do not have a semantic label. If the image is for decoration only and is not relevant to the good functionality of the application, it can be ignored.

Image(
                image: NetworkImage (
                  '...',
                ),
                semanticLabel: 'Image with people learning to sing',
           );

============ Icons ===============

By default it does not have a semantic label, if the icon is only decorative it can be ignored

**Optional**

Icon(
                Icons.remove,
                semanticLabel: 'Remove item from card',
              ),

============ Buttons ===============

By default it is labeled "Button", this can be confused for users who navigate the application with a screen reader.

Semantics(
                      label: "Go to the shopping cart page",
                      child: ... (
                        onPressed: () {},
                        child: Text ('Show shopping cart'),
                      ),
                    ),

Where ... Button is some widget that ends in Button.

Button Exceptions:

BackButton, CloseButton with a semantic label "Back Button" and "Close Button".

============ Contrast ================

We can see in the following link the process to calculate the contrast:
https://www.w3.org/WAI/WCAG21/Techniques/general/G17.html

and if the process to calculate the contrast ratio is very complex, we can suggest to the developer that he can use the external tools that calculate this contrast color like the following: https://www.usableyaccesible.com/recurso_misvalidadores.php#color

You can use libraries to know the main color of images and the Theme.of(context) to know the background color.

Suggest in the following cases:

Container(
   color: [someColor],
   child: Text () or Image ()
);

Container(
   decoration: BoxDecoration (
        color: [someColor],
   ),
   child: Text () or Image ()
);

...Button(
     color: [someColor],
  );

Scaffold(
body: Center (
    child: Text () or Image (),
  ),
);

more info in:
https://www.w3.org/TR/WCAG21/#contrast-minimum
https://www.w3.org/TR/WCAG21/#contrast-enhanced

**Comment by srawlins** [comment]
Yes, please file individual issues. Each lint rule might have a different priority from another, and it makes it easier to track completed work. It also allows for specific conversations regarding the intricacies of the different rules.
